src/main/java/org/jboss/tm/usertx/TransactionListener.java

```diff
@@ -0,0 +1,16 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.Transaction;
+
+/**
+ * A listener callback for transaction related events
+ */
+public interface TransactionListener {
+    /**
+     * The callback for notifying registered listeners of a transaction related event
+     *
+     * @param transaction the transaction that the caller is registering interest in
+     * @param transactionEvent indication of what kind of change has occurred
+     */
+    void stateChange(Transaction transaction, TransactionEvent transactionEvent);
```

Comment:
The listener registration interface is void addListener (Transaction transaction, TransactionListener listener) so the callback must reference which transaction the callback is for. In the future we may add events for when the txn isn't bound so TM.getTransaction() will not give him the transaction back.

As to putting the txn into TransactionEvent in some future JIRA:- that would not work because enums cannot be extended.